### ISSUE TITLE
Tasker Integration - Add / Remove Favourites

### ISSUE THREAD

**Issue by folkg** [8bitPit/Niagara-Issues#786]
## We're getting a lot of duplicate issues.
**Please use GitHub's search function to locate existing tickets and contribute to these instead of opening another one. Most common features which are implemented in other launchers are requested already.**

---

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by folkg** [comment]
If you could use Tasker conditions to add / remove apps from favourites you could create your own kind of smart app suggestions. If you are at work, you could add app x, y, z to favourites. If you are at home, you could remove x, y, z and add a, b ,c. Similar if you are driving. If you have a certain calendar event, you could add a certain app to the favourites.

**Comment by Bushmills** [comment]
Other actions could be:
- set text colour of specific launcher item
- show/hide letter bar
- hide/unhide app
- turn on/off start screen notifications

**Comment by 8bitPit** [comment]
Controlling your favorites with tasker sounds great. Besides the upcoming inbuilt Media & Car category you could create your own rules when an app should show up. Hope you can understand that the integration will be a pro feature, I believe the feature is necessary, but would a cool enhancement to Niagara. @Bushmills the tweaks you’re describing sound more minor to me. They’re also neat, but I would say they are on lower priority.

**Comment by Bushmills** [comment]
@8bitPit: Yes, minor in terms of effort to implement, I hope. But useful: Changing item colours resulting from some  events are great indicators, without cluttering the screen with all kinds of stuff. I do use those with Lightning Launcher which provides Locale API (which tasker is using too, also Automagic which I'm using), for a handful of things. For example, proper "home/away" detection - I have a small reliability issue with this, so colouring the nav icon or label accordingly allows me to verify on a glimpse. Some modal stuff, like activating the proximity sensor for a specific action by gesture (of which I want some indicator too), charge/discharge (though I'm not using an item label for this) and some more. I think charme of colour change comes just from this  being a very simple indicator.
I've removed font size change of individual items prior to publishing thinking that may be a too involved action to interface.

**Comment by 8bitPit** [comment]
Thanks for the explanation @Bushmills. I can understand now why changing colors for example would be neat. I'll see what I can do and let you know here when I start using their APIs.

**Comment by Martin5001** [comment]
As to what you mentioned in one of the issues linked above:

> We plan to make this possible by integrating Tasker or other automation apps like Automate

I personally would prefer Automate over Tasker. I think Automate has much better UI and is easier to use. But of course someone might disagree – so it might be good to vote on this?

**Comment by 8bitPit** [comment]
@Martin5001 luckily Automate and Tasker support the same API :)

**Comment by Bushmills** [comment]
As does Automagic, and supposedly a whole bunch of automation apps.

**Comment by stephenwilley** [comment]
+1 for this - it'd be great to be able to geofence favourite apps for example

**Comment by josevega96** [comment]
Would really love this feature :), btw thanks for all you hard work

**Comment by Manu99it** [comment]
Would really love this! Also I would really like to have the possibility to hide the custom widget when certain conditions are met: for example when the media widget appears

**Comment by IanDTrapp** [comment]
I would love this to be able to show and hide my work apps during the work day. Glad to see this is gaining traction!

**Comment by sandeshdroid** [comment]
Niagara is about to become a full blown launcher. I think it's time to integrate with tasker. Tasker has lots of context. Fews to mention:
- When I get notification, i can make it favorite. So, I can see app and notification without searching.
- When I am at home, I want youtube and netflix to be my favorite.
- When I am at office, I want autocad, gps, unit conversion app to be my favorite.(only weekdays)
- If internet is connected, in my home, then make browser favorite.
- and many more.
I thinks colour changing with tasker is unnecessary since tasker itself can change accent of phone in android 12. You can make red accent for office and blue for home from tasker.

**Comment by Bushmills** [comment]
My suggestion is to get inspiration from Lightning Launcher for an idea of what range of functionally could me made controllable through Locale API. That launcher has the most comprehensive support of any launcher I've been looking at., beating Total launcher (which provides only token support) by far.

I did write a short series of articles about launcher and automation app integration, which also show (but don't demonstrate all functions - there are too many) the extent of support that launcher offers. These articles have been published and are available online, as a possible substitute for actually looking at the launcher, if interested.

Btw, Lightning Launcher is still my launcher of choice, for exactly this reason.

**Comment by sandeshdroid** [comment]
I loved Lightning Launcher and TUI launcher(text based launcher, stored all parameters in file, just edit file with any automation app). Now, I want things that works with little effort. Anyway I am exiced to see what Niagara can do with automation.

**Comment by Bushmills** [comment]
I think the "trick" with an automation app supported launcher is simply: bind as many as possible launcher capabilities to the automation API, and let the user decide which ones he/she will actually invoke. An function made available but remaining unused hurts less than an unavailable one which the user intended to use.

**Comment by sandeshdroid** [comment]
Please follow send intent approach. This have two benefits: no ui for tasker is necessary and any app(like shell, broadcast app) can do it. This is how [Notify for MI Band](http://www.mibandnotify.com/help/tasker_send_intent.php) does this.

**Comment by sandeshdroid** [comment]
Now this is 3 yrs old and may be no progress at all. Is it possible to make a separate foss 'niagara launcher tasker plugin' app, so dev and community both can work together. It will help to make niagara app minimal, bugs free if any to innocent non tasker niagara user. The app can be distributed through github, so no confusion or negative review bombing on playstore.

**Comment by Developer-Mike** [comment]
+1 This would be an awesome addition. Especially because the Niagara Launcher focuses on simplicity and cleanness. It would allow anyone to customise the launcher without adding any unnecessary UI elements to the launcher itself.

**Comment by CheariX** [comment]
Unfortunately, this features seems to receive the lowest priority. At least, we haven‘t got any information about its progress for a long time. It looks like Niagara is currently focusing more on appearance features (icons, wallpapers, clock styles).

**Comment by Oddward** [comment]
> Unfortunately, this features seems to receive the lowest priority. At least, we haven‘t got any information about its progress for a long time. It looks like Niagara is currently focusing more on appearance features (icons, wallpapers, clock styles).

One of the reasons these have all been released now is because they were already being developed for a long time at different times, so having a customization themed release cycle made it fruitful (along with addressing a number of customization pain points at once). 

This year should be focused elsewhere, like quality of life updates and hopefully some other main demands. It's another major feature so it'd require blocking out some time for it.

For now, I suggest advocates highlight exactly what kind of things they'd like to do with the integration to provide more context, and upvote.

**Comment by damElo850** [comment]
Here are my suggestions for the plugin:

Events:
✅Niagara button tapped,moved up or held down🛠️example usage🛠️ tapping Niagara button turns on/off light bulb in room(with home assistant plugin) or activates performance mode or kills background apps

Actions:
✅Change wallpaper/icon pack, theme,fonts color🛠️example usage🛠️ change wallpaper+theme+icon pack based on time of day or weather 
✅Action to launch shortcuts static and dynamic(from apps) created in Niagara 🛠️example usage🛠️ action to launch shortcut of radio with favorite station or favorite discord server or specific conversation/chat on Messenger/Whatsapp/telegram etc.

More about dynamic shortcuts:
https://www.reddit.com/r/tasker/comments/gkounp/pluginbeta_taskerlaunchershortcut/

Simple tutorial how to add plugin to tasker in 5 minutes:
https://youtu.be/48IVJgDtu6Y

**Comment by stephenwilley** [comment]
My suggestion for actions would be as the original title suggests: Add / Remove Favourites

As @folkg and @8bitPit pointed out originally, being able to update that 'front page' list depending on whether you're at work, or some other Tasker based trigger, would be great and allow a user-controllable extension of the 'Bluetooth is connected so show the audio player' style behaviour.

**Comment by TFackler** [comment]
Someone also requested changing the favorites based on the device mode (e.g. bedtime mode). That should be doable using Tasker, but might also be great as a standalone trigger for non-power users.

**Comment by CheariX** [comment]
We are close to the sixth year's anniversary of this issue 😄. 

Any chance to get an update on this issue's current state? 
Is anyone working on it? 

Do you think we can expect some Tasker integration (e.g., https://github.com/8bitPit/Niagara-Issues/issues/786#issuecomment-1885576693) by 2025?
